helm-charts/splunk-otel-collector/values.yaml

```diff
@@ -1022,3 +1022,269 @@ service:
+      # - tcp.syn_timeouts
+      # - tcp.new_sockets
+      # - tcp.resets
+      ### udp ###
+      # - udp.bytes
+      # - udp.packets
+      # - udp.active
+      # - udp.drops
+      ### dns ###
+      # - dns.client.duration.average
+      # - dns.server.duration.average
+      # - dns.active_sockets
+      # - dns.responses
+      # - dns.timeouts
+      ### http ###
+      # - http.client.duration.average
+      # - http.server.duration.average
+      # - http.active_sockets
+      # - http.status_code
+      ### ebpf_net ###
+      # - ebpf_net.span_utilization_fraction
+      # - ebpf_net.pipeline_metric_bytes_discarded
+      # - ebpf_net.codetiming_min_ns
+      # - ebpf_net.entrypoint_info
+      # - ebpf_net.otlp_grpc.requests_sent
+      # - ebpf_net.connections
+      # - ebpf_net.rpc_queue_elem_utilization_fraction
+      # - ebpf_net.disconnects
+      # - ebpf_net.codetiming_avg_ns
+      # - ebpf_net.client_handle_pool
+      # - ebpf_net.otlp_grpc.successful_requests
+      # - ebpf_net.span_utilization
+      # - ebpf_net.up
+      # - ebpf_net.rpc_queue_buf_utilization_fraction
+      # - ebpf_net.collector_log_count
+      # - ebpf_net.time_since_last_message_ns
+      # - ebpf_net.bpf_log
+      # - ebpf_net.codetiming_count
+      # - ebpf_net.message
+      # - ebpf_net.otlp_grpc.bytes_sent
+      # - ebpf_net.pipeline_message_error
+      # - ebpf_net.pipeline_metric_bytes_written
+      # - ebpf_net.codetiming_max_ns
+      # - ebpf_net.span_utilization_max
+      # - ebpf_net.client_handle_pool_fraction
+      # - ebpf_net.span_utilization_fraction
+      # - ebpf_net.rpc_latency_ns
+      # - ebpf_net.agg_root_truncation
+      # - ebpf_net.clock_offset_ns
+      # - ebpf_net.otlp_grpc.metrics_sent
+      # - ebpf_net.otlp_grpc.unknown_response_tags
+      # - ebpf_net.collector_health
+      # - ebpf_net.codetiming_sum_ns
+      # - ebpf_net.otlp_grpc.failed_requests
+      # - ebpf_net.rpc_queue_buf_utilization
+    telemetryPort: 7000
+    statsPromPort: 7001
+    image:
+      name: splunk-network-explorer-reducer
+      # tag: latest-v0.9
+      # repository: quay.io/signalfx
+    resources: {}
+    nodeSelector: {}
+    affinity: {}
+    tolerations: []
+
+  rbac:
+    create: true
+
+  podSecurityPolicy:
+    enabled: true
+    annotations: {}
+      ## Specify pod annotations
+      ## Ref: https://kubernetes.io/docs/concepts/policy/pod-security-policy/#apparmor
+      ## Ref: https://kubernetes.io/docs/concepts/policy/pod-security-policy/#seccomp
+      ## Ref: https://kubernetes.io/docs/concepts/policy/pod-security-policy/#sysctl
+      ##
+      # seccomp.security.alpha.kubernetes.io/allowedProfileNames: '*'
+      # seccomp.security.alpha.kubernetes.io/defaultProfileName: 'docker/default'
+      # apparmor.security.beta.kubernetes.io/defaultProfileName: 'runtime/default'
+
+  # otel collector in gateway configuration for network explorer.
+  # these values override those specified above.
+  gateway:
```

Comment:
Why do we need another section for the gateway? Can't we reuse existing `gateway`?

Reply:
These are used to supplement / overwrite the existing gateway settings.  In the _helpers template you can see the logic for it (the gateway define)  I'm not at all sold on doing it this way, however.  Basically the important things we want to do is set the replicas down to 1, ensure that the gateway is enabled, and especially set the limits lower (based on feedback we've received)  So if you think we ought to decompose this to the individual settings or something, that's fine too.

Reply:
The problem is that we expose the same user configuration interface in two places which makes it confusing and harder to maintain I believe. We can still use one config interface even if different default values applied based on how the gateway is enabled. But it would be better to find some middle ground and have a reasonable default for both use cases.

> set the replicas down to 1
Why is this required? It's not the best choice for HA

> ensure that the gateway is enabled
You're already doing it with the gatewayEnabled helper, right?

> especially set the limits lower (based on feedback we've received)
The gateway is recommended to be enabled on big clusters, that's why we have such high limits. We can discuss something in between if you agree.